R/aggregate_game_stats_kicking.R

```diff
@@ -0,0 +1,186 @@
+      fg_blocked_distance = sum(.data$dist[.data$temp_block_idx], na.rm = TRUE),
+      .groups = "drop"
+    )
+
+
+  # Extra points
+  df_pat <- df_fg_or_pat %>%
+    dplyr::filter(.data$extra_point_attempt == 1) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      pat_made = sum(.data$pat_res == "good", na.rm = TRUE),
+      pat_att = sum(.data$extra_point_attempt, na.rm = TRUE),
+      pat_missed = sum(.data$pat_res == "failed", na.rm = TRUE),
+      pat_blocked = sum(.data$pat_res == "blocked", na.rm = TRUE),
+      pat_pct = round(.data$pat_made / .data$pat_att, 3L),
+      .groups = "drop"
+    )
+
+
+  # The Game Winning kicks distance include up to one value at the weekly level
+  # but can include multiple across the season. This is one way to account for that.
+  # the downside is that the column names change depending on if it is weekly vs
+  # seasonal.
+  if (weekly) {
+    gw_dist_name <- "gwfg_distance"
+  } else {
+    gw_dist_name <- "gwfg_distance_list"
+  }
+
+  # See the above note. I wonder if this should also include field goals that tie
+  # the game but I kept the filter dplyr::between(score_differential, -2, 0) the way
+  # that is was previously. If you do include field goals that send the game into OT,
+  # then you'll probably need to include the gwfg_distance AND gwfg_distance_list columns
+  # in the weekly data
+  game_winners <- df_fg_or_pat %>%
+    dplyr::group_by(.data$game_id, .data$team) %>%
+    dplyr::filter(.data$fixed_drive == max(.data$fixed_drive, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$field_goal_attempt == 1, dplyr::between(.data$score_differential, -2, 0)) %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::summarise(
+      gwfg_att = dplyr::n(),
+      !!gw_dist_name := if (weekly) .data$dist else paste(na.omit(.data$dist), collapse = ";"),
+      gwfg_made = sum(.data$fg_res == "made", na.rm = TRUE),
+      gwfg_missed = sum(.data$fg_res == "missed", na.rm = TRUE),
+      gwfg_blocked = sum(.data$fg_res == "blocked", na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  # Prepping data to merge-in player names
+  df_player_names <- nflreadr::load_players() %>%
+    dplyr::select(
+      "player_id" = "gsis_id",
+      "player_display_name" = "display_name",
+      "player_name" = "short_name",
+      "position",
+      "position_group",
+      "headshot_url" = "headshot"
+    )
+
+  # Joining all the data together and organizing the first few columns.
+  # I was not sure what order you want the columns
+  full_kicks <- df_field_goals %>%
+    dplyr::full_join(df_pat, as.character(grp_vars)) %>%
+    dplyr::full_join(game_winners, as.character(grp_vars)) %>%
+    dplyr::left_join(df_player_names, "player_id") %>%
+    dplyr::group_by(!!!grp_vars) %>%
+    dplyr::mutate(games = dplyr::n()) %>%
+    dplyr::ungroup() %>%
+    dplyr::select(
+      "season", dplyr::any_of("week"), dplyr::any_of("season_type"), "player_id",
```

Comment:
```suggestion
      dplyr::any_of(c("season", "week","season_type")), "player_id",
```

consistent with the logic above